workflow-examples/src/test/java/com/redhat/parodos/examples/integration/KubeapiWorkFlow.java

```diff
@@ -0,0 +1,130 @@
+
+	private final String workflowName = "kubeapiWorkFlow";
+
+	private final String taskName = "kubeapiTask";
+
+	@Test
+	public void runFlow() throws ApiException {
+		ApiClient defaultClient = Configuration.getDefaultApiClient();
+
+		defaultClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, "Basic " + CredUtils.getBase64Creds("test", "test"));
+
+		ProjectApi apiInstance = new ProjectApi(defaultClient);
+
+		try {
+			// RETRIEVE ALL PROJECTS AVAILABLE
+			List<ProjectResponseDTO> projects = apiInstance.getProjects();
+
+			// CHECK IF project ALREADY EXISTS
+			ProjectResponseDTO testProject = projects.stream()
+					.filter(prj -> projectName.equals(prj.getName()) && projectDescription.equals(prj.getDescription()))
+					.findAny().orElse(null);
+
+			// CREATE PROJECT IF NOT EXISTS
+			if (testProject == null) {
+				// DEFINE A PROJECT REQUEST
+				ProjectRequestDTO projectRequestDTO = ProjectRequestDTO.builder().name(projectName)
+						.description(projectDescription).build();
+
+				testProject = apiInstance.createProject(projectRequestDTO);
+			}
+
+			// ASSERT PROJECT IS PRESENT
+			projects = apiInstance.getProjects();
+			assertNotNull(projects.stream()
+					.filter(prj -> projectName.equals(prj.getName()) && projectDescription.equals(prj.getDescription()))
+					.findAny().orElse(null));
+
+			// GET workflow DEFINITIONS
+			WorkflowDefinitionApi workflowDefinitionApi = new WorkflowDefinitionApi();
+			List<WorkFlowDefinitionResponseDTO> simpleSequentialWorkFlowDefinitions = workflowDefinitionApi
+					.getWorkFlowDefinitions(workflowName);
+			assertEquals(1, simpleSequentialWorkFlowDefinitions.size());
+
+			// GET WORKFLOW DEFINITION BY Id
+			WorkFlowDefinitionResponseDTO simpleSequentialWorkFlowDefinition = workflowDefinitionApi
+					.getWorkFlowDefinitionById(simpleSequentialWorkFlowDefinitions.get(0).getId().toString());
+
+			// EXECUTE WORKFLOW
+			WorkflowApi workflowApi = new WorkflowApi();
+
+			// 1 - Define WorkRequests
+			WorkRequestDTO workGet = WorkRequestDTO.builder().workName(taskName)
+					.arguments(Arrays.asList(new ArgumentRequestDTO().key("api-group").value(""),
+							new ArgumentRequestDTO().key("api-version").value("v1"),
+							new ArgumentRequestDTO().key("kind-plural-name").value("configmaps"),
+							new ArgumentRequestDTO().key("operation").value("get"),
+							new ArgumentRequestDTO().key("resource-name").value("my-cm"),
+							new ArgumentRequestDTO().key("resource-namespace").value("test"),
+							new ArgumentRequestDTO().key("work-ctx-key").value("theKey")))
+					.build();
+
+			WorkRequestDTO workCreate = WorkRequestDTO.builder().workName(taskName).arguments(Arrays.asList(
+					new ArgumentRequestDTO().key("api-group").value(""),
+					new ArgumentRequestDTO().key("api-version").value("v1"),
+					new ArgumentRequestDTO().key("kind-plural-name").value("configmaps"),
+					new ArgumentRequestDTO().key("operation").value("create"),
```

Comment:
Where is the identity of the k8s client? How do we verify if the user is authorized to create configmaps in k8s?

Reply:
kubeconfig obviously